File: java/common/src/main/java/com/ais/middleware/common/events/prs/UWDeterminationRequestedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by StatusCode6UWSaga to trigger parallel UW assignment determination.
 * Topic: prs.events.uw-determination-requested
 *
 * ⚠️ DEMO GAP: UW determination rules not confirmed — routing logic (UA vs UST)
 * is inferred from inspectionTypeCode (A→UA, B→UST, I/J→UA).
 * Real rule codes need to be provided by PRS team.
 */
public record UWDeterminationRequestedEvent(
        String correlationId,
        String appraisalId,
        String policyNumber,
        String inspectionTypeCode,
        String producerControlCode,
        OffsetDateTime requestedAt
) {
    @JsonCreator
    public UWDeterminationRequestedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("appraisalId") String appraisalId,
            @JsonProperty("policyNumber") String policyNumber,
            @JsonProperty("inspectionTypeCode") String inspectionTypeCode,
            @JsonProperty("producerControlCode") String producerControlCode,
            @JsonProperty("requestedAt") OffsetDateTime requestedAt
    ) {
        this.correlationId = correlationId;
        this.appraisalId = appraisalId;
        this.policyNumber = policyNumber;
        this.inspectionTypeCode = inspectionTypeCode;
        this.producerControlCode = producerControlCode;
        this.requestedAt = requestedAt;
    }
}

File: dotnet/Middleware.Contracts/Events/UWDeterminationRequestedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — saga requests parallel UW determination
public sealed class UWDeterminationRequestedEvent
{
    public string AppraisalId { get; set; } = string.Empty;
    public string PolicyNumber { get; set; } = string.Empty;
    public string ProducerCode { get; set; } = string.Empty;

    // InspectionTypeCode determines UA vs UST routing
    // Known values from spec: A=Appraisal, B=Appraisal UST, I=Inspection, J=Inspection
    // ⚠️ DEMO GAP: Actual rule codes not confirmed with PRS team — REPLACE_ME_UW_RULE_CODES
    public string InspectionTypeCode { get; set; } = string.Empty;

    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset RequestedAt { get; set; }
}
